### PR TITLE
file-roller: fixed fatal error occuring when extracting zip files

### PR DESCRIPTION
**Pull request by dasj19**
###### Motivation for this change
Check: https://github.com/NixOS/nixpkgs/pull/117081#issuecomment-827193689

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120934` at 44d004d7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>gnome3.file-roller</li>
    <li>pantheon.extra-elementary-contracts</li>
  </ul>
</details>

File: pkgs/desktops/gnome-3/apps/file-roller/default.nix [hunk 1]
```diff
@@ -24,6 +24,11 @@ stdenv.mkDerivation rec {
     patchShebangs data/set-mime-type-entry.py
   '';
 
+  postFixup = ''
+    wrapProgram "$out/bin/file-roller" \
+      --prefix PATH : ${lib.makeBinPath [ unzip ]}
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/apps/file-roller/default.nix:29]: How is cpio invoked? Won’t it be needed here as well?

File: pkgs/desktops/gnome-3/apps/file-roller/default.nix [hunk 2]
```diff
@@ -24,6 +24,11 @@ stdenv.mkDerivation rec {
     patchShebangs data/set-mime-type-entry.py
   '';
 
+  postFixup = ''
+    wrapProgram "$out/bin/file-roller" \
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/apps/file-roller/default.nix:28]: Please add a comment that this is workaround for https://gitlab.gnome.org/GNOME/file-roller/-/issues/40.

Comment (dasj19) [reply pkgs/desktops/gnome-3/apps/file-roller/default.nix:29]: done, thanks for review

File: pkgs/desktops/gnome-3/apps/file-roller/default.nix (hunk 1 above)
Comment (dasj19) [reply pkgs/desktops/gnome-3/apps/file-roller/default.nix]: fixed. thanks.

File: pkgs/desktops/gnome-3/apps/file-roller/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/desktops/gnome-3/apps/file-roller/default.nix:29]: Looks like cpio is not necessary here if we use the patch I accidentally pushed https://gitlab.gnome.org/GNOME/file-roller/-/commit/af27465a3ef07622e89890ef4b1aacbb754b6fb9

Comment (dasj19) [reply pkgs/desktops/gnome-3/apps/file-roller/default.nix:39]: Patch added. Thanks

**Comment by jtojnar** [comment]
Great, thanks.

**Pull request merged**